pkg/controller/replication/replication_controller.go

```diff
@@ -195,6 +198,13 @@ func (rm *ReplicationManager) Run(workers int, stopCh <-chan struct{}) {
 // getPodController returns the controller managing the given pod.
 // TODO: Surface that we are ignoring multiple controllers for a single pod.
 func (rm *ReplicationManager) getPodController(pod *api.Pod) *api.ReplicationController {
+	// look up in the cache, if cached and the cache is valid, just return cached value
+	controller, cached := rm.lookupCache.GetCachedRC(pod)
+	if cached && rm.isCacheValid(pod, controller) {
```

Comment:
BTW - it seems that this PR solves the issue for RC, #21674 is a more generic parallelization which we can use in other places too).
[Anyway - I think we should proceed with this one too, but maybe not for 1.2?]